covid19-dashboard/dc-covid19/src/DataCalculator.ts

```diff
@@ -1,90 +1,126 @@
             if (!(deltaDaysFromIterativeDate in data) || !(geoId in data[deltaDaysFromIterativeDate])) continue
+
             // Get the values for both days
-            const iterativeDateValue = data[iterativeDate][geoId]
+            const iterativeDateValue = data[date][geoId]
             const deltaDaysFromIterativeDateValue = data[deltaDaysFromIterativeDate][geoId]
-            // If one of the values is invalid, or falsy, continue.
-            if (!iterativeDateValue || !deltaDaysFromIterativeDateValue) continue
-            // Do some calculation on the data using the calculation function.
-            let result: number | null = null;
 
-            switch (calculationType){
-                case 'absolute':
-                    result = iterativeDateValue
-                    break;
-                case 'difference':
-                    result = iterativeDateValue - deltaDaysFromIterativeDateValue
-                    break;
-                case 'perCapita':
-                    result = ((iterativeDateValue - deltaDaysFromIterativeDateValue) / geoIdToPopulation[geoId]) * 10000
-                    break;
-                case 'increase':
-                    result = (iterativeDateValue / deltaDaysFromIterativeDateValue) - 1
-                    break;
-                case 'absolutePerCapita':
-                    result = (iterativeDateValue / geoIdToPopulation[geoId]) * 10000
-                    break;
-            }
+            // Do some calculation on the data using the getResult function.
+            const result: number | null = getResult(
+                [deltaDaysFromIterativeDateValue, iterativeDateValue],
+                geoIdToPopulation?.[geoId],
+                calculationType)
 
-            // If the result is valid, store it
-            if (result) {
+            // If the result is valid, store it otherwise, skip this specific value.
+            if (result !== undefined && result !== null) {
                 // If this is the first time storing this date in the output, make some room for it.
-                if (!(iterativeDate in outputData)) outputData[iterativeDate] = {}
-                outputData[iterativeDate][geoId] = result
+                if (!(date in outputData)) outputData[date] = {}
+                outputData[date][geoId] = result
             }
         }
-        // Add one to the iterativeDate. Example: from "2020-01-01" to "2020-01-02"
-        iterativeDate = addOrSubtractNDaysToDate(iterativeDate, 1)
+        // Add one day to the iterativeDate. Example: from "2020-01-01" to "2020-01-02".
+        date0 = date0.add(1, "days")
     }
     return outputData
+}
+
+/**
+ * Given two values (or one if absolute or absolutePerCapita), return the result of calculationType.
+ * @param values: [number, number] -> values[0] is date - deltaDays, values[1] is the actual date.
+ * @param population: some calculations require the population.
```

Comment:
also take the population out from here. So this function do binary operations on values

Reply:
This function no longer exists.